synapse/lib/stormsvc.py

```diff
@@ -139,23 +140,37 @@ async def _runSvcInit(self):
         self.core.svcsbysvcname[self.svcname] = self
 
         await self.core.feedBeholder('svc:set', {'name': self.name, 'iden': self.iden, 'svcname': self.svcname, 'version': self.svcvers})
+        # if the old service is the same as the new service, just skip
 
-        try:
-            await self.core._delStormSvcPkgs(self.iden)
-
-        except asyncio.CancelledError:  # pragma: no cover  TODO:  remove once >= py 3.8 only
-            raise
-
-        except Exception:
-            logger.exception(f'_delStormSvcPkgs failed for service {self.name} ({self.iden})')
+        oldpkgs = self.core.getStormSvcPkgs(self.iden)
+        byname = {}
+        done = set()
+        for pdef in oldpkgs:
+            iden = s_hashitem.hashitem(pdef)
+            byname[pdef.get('name')] = iden
 
         # Register new packages
         for pdef in self.info.get('pkgs', ()):
+            pdef['svciden'] = self.iden
```

Comment:
probably need this to still be in the `try` block to handle malformed pkg definitions